services/venice-controller/src/main/java/com/linkedin/venice/controller/init/ControllerClientBackedSystemSchemaInitializer.java

```diff
@@ -0,0 +1,246 @@
+                "Error when updating system store " + storeName + " in cluster " + clusterName
+                    + " after retries. Error: " + updateStoreResponse.getError());
+          }
+          LOGGER.info("System store {} has been created.", storeName);
+        }
+      } else {
+        throw new VeniceException(
+            "Error when getting system store " + storeName + " from cluster " + clusterName + " after retries. Error: "
+                + storeResponse.getError());
+      }
+    }
+  }
+
+  private void checkIfKeySchemaMatches(String storeName) {
+    SchemaResponse keySchemaResponse =
+        controllerClient.retryableRequest(DEFAULT_RETRY_TIMES, c -> c.getKeySchema(storeName));
+    if (keySchemaResponse.isError()) {
+      throw new VeniceException(
+          "Error when getting key schema from system store " + storeName + " in cluster " + clusterName
+              + " after retries. Error: " + keySchemaResponse.getError());
+    }
+    Schema curKeySchema = AvroCompatibilityHelper.parse(keySchemaResponse.getSchemaStr());
+    if (!curKeySchema.equals(keySchema)) {
+      LOGGER.error(
+          "Key Schema of {} in cluster {} is already registered but it is INCONSISTENT with the local definition.\n"
+              + "Already registered: {}\n" + "Local definition: {}",
+          storeName,
+          clusterName,
+          curKeySchema.toString(true),
+          keySchema.toString(true));
+    }
+  }
+
+  private void checkAndMayRegisterValueSchema(
+      String storeName,
+      int valueSchemaId,
+      Schema schemaInZk,
+      Schema schemaInLocalResources) {
+    if (schemaInZk == null) {
+      SchemaResponse addValueSchemaResponse = controllerClient.retryableRequest(
+          DEFAULT_RETRY_TIMES,
+          c -> c.addValueSchema(storeName, schemaInLocalResources.toString(), valueSchemaId));
+      if (addValueSchemaResponse.isError()) {
+        throw new VeniceException(
+            "Error when adding value schema " + valueSchemaId + " to system store " + storeName + " in cluster "
+                + clusterName + " after retries. Error: " + addValueSchemaResponse.getError());
+      }
+      LOGGER.info("Added new schema v{} to system store {}.", valueSchemaId, storeName);
+    } else {
+      if (schemaInZk.equals(schemaInLocalResources)) {
+        LOGGER.info(
+            "Schema v{} in system store {} is already registered and consistent with the local definition.",
+            valueSchemaId,
+            storeName);
+      } else {
+        LOGGER.warn(
+            "Schema v{} in system store {} is already registered but it is INCONSISTENT with the local definition.\n"
+                + "Already registered: {}\n" + "Local definition: {}",
+            valueSchemaId,
+            storeName,
+            schemaInZk.toString(true),
+            schemaInLocalResources.toString(true));
+      }
+    }
+  }
+
+  private void checkAndMayRegisterWriteComputeSchema(
```

Comment:
nit: writeCompute ->partialUpdate, would like to standardize the feature naming here.

Reply:
The converter is still called `WriteComputeSchemaConverter`. Is there any ongoing progress to change all writeCompute to partialUpdate?